### ISSUE TITLE
streamline config/storage API in terms of pointer usage

### ISSUE THREAD

**Issue by GlenDC** [zero-os/0-Disk#542]
We are not consistent in how we create, pass and use `config.StorageServerConfig` and `config.StorageClusterConfig`. Sometimes they are created and passed on by value, and other times as pointers. Even worse, often they go (multiple times) from pointer to value and perhaps even back to pointer.

+ The `Validate() error` method for those types should return an error in case the pointer version of that type is `nil`, this will help a lot;

After that I'm not too sure if we should either use them by value, or by pointer. I think using them by value and passing them by value is probably a lot safer, and in general it's also not too expensive.

+ `config.StorageClusterConfig` just contains a slice;
+ `config.StorageServerConfig` contains a string, int64 and uint8, so still manageable, especially as it's not really passed frequently in any performance critical location;

For now I leave this on the beta-1 backlog, but this can and should probably be done right after #483 is resolved and merged into master.
